jetson/odrive_bridge/src/__main__.py

```diff
@@ -258,28 +251,25 @@ def on_event(self, event):
         publish_state_msg(state_msg, odrive_bridge.get_state())
 
     def update(self):
-        try:
-            errors = modrive.check_errors()
-            modrive.watchdog_feed()
-        except Exception:
-            errors = 0
-            lock.acquire()
-            self.on_event("disconnected odrive")
-            lock.release()
-            print("unable to check errors of unplugged odrive")
+        if (str(self.state) == "ArmedState"):
+            try:
+                errors = modrive.check_errors()
+                modrive.watchdog_feed()

```

Comment:
Don't you need to declare modrive and lock as a global to reference them here? Aren't you just creating it locally rn?

Reply:
no, arguably the only place you need it declared globally is in connect when it's assigned to, if it's read-only in a function python assumes it's part of the global scope. it's probably declared globally everywhere else bc the original authors were being careful if i had to guess. also update is in a lock already